provider/dnsimple/dnsimple.go

```diff
@@ -99,7 +99,14 @@ const (
 
 // NewDnsimpleProvider initializes a new Dnsimple based provider
 func NewDnsimpleProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provider.ZoneIDFilter, dryRun bool) (provider.Provider, error) {
+	return BuildDnsimpleProvider(domainFilter, zoneIDFilter, dryRun)
+}
+
+// Create a new Dnsimple based provider returning a *dnsimpleProvider. The *dnsimpleProvider return type is needed for testing purposes
+// therefore this method, and not NewDnsimpleProvider, must be the one used by dnsimple_test.go
+func BuildDnsimpleProvider(domainFilter endpoint.DomainFilter, zoneIDFilter provider.ZoneIDFilter, dryRun bool) (*dnsimpleProvider, error) {
 	oauthToken := os.Getenv("DNSIMPLE_OAUTH")
+	dnsimpleAccountId := os.Getenv("DNSIMPLE_ACCOUNT_ID")
```

Comment:
```suggestion
```

Reply:
Added in https://github.com/kubernetes-sigs/external-dns/pull/4274/commits/487501d923a59cb9e93ec3ebf4ef5447476540a7